packages/faency/src/components/InputTags.tsx

```diff
@@ -0,0 +1,182 @@
+import { InputProps } from 'mdlz-prmtz'
+import styled from 'styled-components'
+import { theme } from '../theme'
+import { Box } from './Box'
+
+const StyledInput = styled('input')`
+  border: none;
+  background: none;
+  outline: none;
+  flex-grow: 1;
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  max-height: ${theme.sizes[3]};
+  margin: 0 ${theme.space[1]} ${theme.space[1]} 0;
+`
+
+const Container = styled('div')<{ hasFocus: boolean }>`
+  position: relative;
+  padding: ${theme.space[1]} 0 0 ${theme.space[1]};
+  transition: box-shadow 360ms cubic-bezier(0.4, 0, 0.2, 1);
+  box-sizing: border-box;
+  border-radius: ${theme.radii[4]};
+  box-shadow: 0 0 0 1px ${theme.colors.gray};
+  display: flex;
+  flex-wrap: wrap;
+
+  ${({ hasFocus }) =>
+    hasFocus &&
+    `
+    box-shadow: 0 0 0 2px ${theme.colors.black};    
+  `}
+`
+
+const ChipsContainer = styled('div')`
+  display: flex;
+  flex-wrap: wrap;
+
+  &:first-child {
+    justify-items: flex-end;
+    max-width: 60%;
+  }
+
+  > * {
+    margin: 0 ${theme.space[1]} ${theme.space[1]} 0;
+  }
+`
+
+const SelectContainer = styled(Box)`
+  position: absolute;
+  left: 0;
+  right: 0;
+  top: 100%;
+  margin-top: 3px;
+  background: ${theme.colors.white};
+  border-radius: 2px;
+  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.3);
+  z-index: 10;
+`
+
+const SelectItem = styled(Box)`
+  padding: ${theme.space[1]};
+  transition: all 100ms ease-in;
+
+  &:hover {
+    background: ${theme.colors.blue};
+    color: ${theme.colors.white};
+    cursor: pointer;
+  }
+`
+
+type CustomInputType = InputProps & {
+  onEnter?: (value: string) => void
+}
+
+const CustomInput = ({ onEnter, ...props }: CustomInputType) => {
+  const [value, setValue] = useState(props.value as string)
+
+  const handlePressEnter = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter' && onEnter) {
+      onEnter(value)
+    }
+  }
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value)
+    if (props.onChange) {
+      props.onChange(e)
+    }
+  }
+
+  return <StyledInput {...props} onChange={handleChange} onKeyPress={handlePressEnter} />
+}
+
+type RenderTagType = (tag: string, onDeleteTag: (tag: string) => void) => ReactNode
+
+const defaultRenderTag: RenderTagType = (tag, onDeleteTag) => (
+  <DismissableChip key={tag} dismiss={() => onDeleteTag(tag)}>
+    {tag}
+  </DismissableChip>
+)
+
+type RenderOptionType = (tag: string, onClick: (v: string) => void) => ReactNode
+
+const defaultRenderOption: RenderOptionType = (option, onClick) => (
+  <SelectItem key={option} onClick={() => onClick(option)}>
+    {option}
+  </SelectItem>
+)
+
+interface InputTagsProps {
+  value?: string
+  placeholder?: string
+  tags?: string[]
+  options?: string[]
+  maxTags?: number
+  maxInlineTags?: number
+  renderTag?: RenderTagType
+  renderOption?: RenderOptionType
+  onChange?: (value: string) => void
+  onSubmit?: (value: string) => void
```

Comment:
A think that the value of this component should be the array of selected tags instead of the input one:
- `onSubmit` wording should be reserved at the form level and could be renamed `onChange` 
- `onChange` could be renamed `onInputChange` so we can keep handling logic externally

Reply:
I agree about the names, I'm going to rename them. 

About returning the array of tags when the input is submitted, I think if we do this, we will lose the use case on the Marketplace where the input is not related to the tags.

I thought we could keep the tags bound to state in the parent component, this way we don't need to send them in a callback, and then I can use the `onChange` listener to determine whether the user has 'submitted' the input by pressing the Enter key.

Does that make sense?